src/api/Feed.js

```diff
@@ -595,17 +596,20 @@ class Feed extends Base {
         const appActivityActivityType = shouldShowAppActivity ? [FILE_ACTIVITY_TYPE_APP_ACTIVITY] : [];
         const taskActivityType = shouldShowTasks ? [FILE_ACTIVITY_TYPE_TASK] : [];
         const versionsActivityType = shouldShowVersions ? [FILE_ACTIVITY_TYPE_VERSION] : [];
+        const commentActivityType = permissions[PERMISSION_CAN_COMMENT] ? [FILE_ACTIVITY_TYPE_COMMENT] : [];
         const filteredActivityTypes = [
             ...annotationActivityType,
             ...appActivityActivityType,
-            FILE_ACTIVITY_TYPE_COMMENT,
+            ...commentActivityType,
             ...taskActivityType,
             ...versionsActivityType,
         ];
 
-        const fileActivitiesPromise = shouldUseUAA
-            ? this.fetchFileActivities(permissions, filteredActivityTypes, shouldShowReplies)
-            : Promise.resolve();
+        const fileActivitiesPromise =
+            // Only fetch when activity types are explicitly stated
+            shouldUseUAA && filteredActivityTypes.length
```

Comment:
Are we checking the length because it's actually possible to get to this part of the code and the filteredActivityTypes array is empty?

Reply:
Yes, if all ff are off and the comments are disabled, it's possible to have an empty array. The api handles that as fetching all types, which is what we don't want